### PR TITLE
fix on_startup operator not executing on startup

### PR DESCRIPTION
**Pull request by imanjra**
## What changes are proposed in this pull request?

Fix `on_startup` not executing on startup due to race condition where on_startup operator are executed before invocation queue is initialized

## How is this patch tested? If it is not, please explain why.

Using the VoxelGPT plugin and verifying that the VoxelGPT panel launches on startup 

## Release Notes

Fix `on_startup` not executing on app startup

### Is this a user-facing change that should be mentioned in the release notes?

-   [ ] No. You can skip the rest of this section.
-   [x] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [x] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

### CODE REVIEW

File: app/packages/operators/src/loader.tsx [hunk 1]
```diff
@@ -4,21 +4,15 @@ import { useEffect, useState } from "react";
...
 
-let startupOperatorsExecuted = false;
-
 async function loadOperators(datasetName: string) {
   registerBuiltInOperators();
   await loadOperatorsFromServer(datasetName);
-  if (!startupOperatorsExecuted) {
-    executeStartupOperators();
-    startupOperatorsExecuted = true;
-  }
```
Comment (benjaminpkane) [inline app/packages/operators/src/loader.tsx:21]: I believe the doc string below is incorrect now? `OperatorCore` now queues start-up operators?

**Review by benjaminpkane — COMMENTED** [review]
I'm not sure I see the race condition. For edification purposes, is there specific logic in `useOperators` that causes this? `getInvocationRequestQueue` creates the queue if needed

**Comment by imanjra** [comment]
`getInvocationRequestQueue` indeed creates a queue, however, the one of the executor (the only one atm) for the queue is initialized by `OperatorInvocationRequestExecutor`. The `OperatorInvocationRequestExecutor` uses `useInvocationRequestQueue` which initializes a subscriber for the queue (primarily updating `requests` state to be picked up by `OperatorInvocationRequestExecutor` reactively). However, the notifying to subscriber is not reactive but rather procedural and is invoked after queue is modified. If a subscriber is added with items already in queue (i.e. OperatorInvocationRequestExecutor) they are not notified of the items in the queue. There are alternate ways to go about solving this by notifying the subscriber on add or subscriber checking the queue when adding itself as a subscriber to the queue.

In summary (race-condition):

1. `loadOperators` populates the queue (and [notifies existing subscribers](https://github.com/voxel51/fiftyone/blob/develop/app/packages/operators/src/operators.ts#L778)) with start-up operators and blocks the mount of `OperatorCore` until it's done
2. Subscriber is initialized on the mount of OperatorInvocationRequestExecutor but does not get notified as the queue is already populated
3. Any further call to `queue.add` will execute

File: app/packages/operators/src/loader.tsx (hunk 1 above)
Comment (imanjra) [reply app/packages/operators/src/loader.tsx:21]: Yah, good point. I will update the doc string below.

**Review by benjaminpkane — APPROVED** [review]
LGTM!

File: app/packages/operators/src/StartupOperatorExecutor.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { useEffect } from "react";
+import { executeStartupOperators } from "./operators";
+
+let startupOperatorsExecuted = false;
+
+export default function StartupOperatorExecutor() {
```
Comment (ritch) [inline app/packages/operators/src/StartupOperatorExecutor.tsx:6]: This seems like a workaround to the actual problem. How can we make the following possible regardless of where or when it happens:

```ts
const queue = getInvocationRequestQueue();
queue.add(new InvocationRequest());
```

**Review by ritch — CHANGES_REQUESTED** [review]
PTAL a look at comment. We can discuss offline if needed.

File: app/packages/operators/src/StartupOperatorExecutor.tsx (hunk 2 above)
Comment (ritch) [reply app/packages/operators/src/StartupOperatorExecutor.tsx:6]: > There are alternate ways to go about solving this by notifying the subscriber on add or subscriber checking the queue when adding itself as a subscriber to the queue.

This

Comment (ritch) [reply app/packages/operators/src/StartupOperatorExecutor.tsx:6]: Its important that `executeOperator()` is essentially possible from any state, thats why we introduced the queue in the first place.

**Comment by imanjra** [comment]
Closing in favour of #4136
